alu_u32/src/lt/mod.rs

```diff
@@ -1 +1,150 @@
+extern crate alloc;
 
+use crate::pad_to_power_of_two;
+use alloc::vec;
+use alloc::vec::Vec;
+use columns::{Lt32Cols, LT_COL_MAP, NUM_LT_COLS};
+use core::iter;
+use core::mem::transmute;
+use valida_bus::MachineWithGeneralBus;
+use valida_cpu::MachineWithCpuChip;
+use valida_machine::{
+    instructions, Chip, Instruction, Interaction, Operands, Word, MEMORY_CELL_BYTES,
+};
+use valida_opcodes::LT32;
+
+use p3_air::VirtualPairCol;
+use p3_field::PrimeField;
+use p3_matrix::dense::RowMajorMatrix;
+use p3_maybe_rayon::*;
+
+pub mod columns;
+pub mod stark;
+
+#[derive(Clone)]
+pub enum Operation {
+    Lt32(Word<u8>, Word<u8>, Word<u8>), // (dst, src1, src2)
+}
+
+#[derive(Default)]
+pub struct Lt32Chip {
+    pub operations: Vec<Operation>,
+}
+
+impl<F, M> Chip<M> for Lt32Chip
+where
+    F: PrimeField,
+    M: MachineWithGeneralBus<F = F>,
+{
+    fn generate_trace(&self, _machine: &M) -> RowMajorMatrix<M::F> {
+        let rows = self
+            .operations
+            .par_iter()
+            .map(|op| self.op_to_row(op))
+            .collect::<Vec<_>>();
+
+        let mut trace =
+            RowMajorMatrix::new(rows.into_iter().flatten().collect::<Vec<_>>(), NUM_LT_COLS);
+
+        pad_to_power_of_two::<NUM_LT_COLS, F>(&mut trace.values);
+
+        trace
+    }
+
+    fn global_receives(&self, machine: &M) -> Vec<Interaction<M::F>> {
+        let opcode = VirtualPairCol::constant(M::F::from_canonical_u32(LT32));
+        let input_1 = LT_COL_MAP.input_1.0.map(VirtualPairCol::single_main);
+        let input_2 = LT_COL_MAP.input_2.0.map(VirtualPairCol::single_main);
+        let output = (0..MEMORY_CELL_BYTES - 1)
+            .map(|_| VirtualPairCol::constant(M::F::ZERO))
+            .chain(iter::once(VirtualPairCol::single_main(LT_COL_MAP.output)))
+            .collect::<Vec<_>>();
+
+        let mut fields = vec![opcode];
+        fields.extend(input_1);
+        fields.extend(input_2);
+        fields.extend(output);
+
+        let receive = Interaction {
+            fields,
+            count: VirtualPairCol::single_main(LT_COL_MAP.is_real),
+            argument_index: machine.general_bus(),
+        };
+        vec![receive]
+    }
+}
+
+impl Lt32Chip {
+    fn op_to_row<F>(&self, op: &Operation) -> [F; NUM_LT_COLS]
+    where
+        F: PrimeField,
+    {
+        let mut row = [F::ZERO; NUM_LT_COLS];
+        let cols: &mut Lt32Cols<F> = unsafe { transmute(&mut row) };
+
+        match op {
+            Operation::Lt32(a, b, c) => {
```

Comment:
nit: would prefer naming `Lt32(dst, src1, src2)`, I was interpreting `c` as the output at first